src/main/java/io/jenkins/plugins/postgresql/PostgreSQLSchemaInitialization.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2020, Jenkins project contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.postgresql;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+public class PostgreSQLSchemaInitialization {
```

Comment:
Why not run an initialization script instead of running one query at a time?

Reply:
Would you suggest something like [this](https://stackoverflow.com/a/1044837/8461040)?

The main reason for not doing that is because that would lead to some complications -> The plugin's config can be changed at any time. Also the plugin does not know that if after the config change, the database has been changed. So the current method allows to check if the schema already exists or not.

The current implementation also does not commit the queries till the end, so it actually runs them together IIUC.

Reply:
I'm all for trying something new :). I've always built it with schema sql scripts that get out of band; in this context, it could be applied when the database is successfully configured. 

Do you want the schema creation to happen with the initial fingerprint persistence?

Reply:
I think the design for when to apply configuration needs more thought from me. I'll comment back, and discuss with what I think is the best approach.

I want that if the database has had fingerprints, then they should be persisted, but if they are present on the local file system, then Jenkins core will handle automatically due to the work we did on migration.